### PR TITLE
Add unit test for sync batch norm under distribution strategy.

### PR DESCRIPTION
**Pull request by qlzh727**
This should address the test coverage in https://github.com/keras-team/keras/pull/18671#issuecomment-1775712652.

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
